src/sentry/snuba/metrics/datasource.py

```diff
@@ -68,17 +90,21 @@ def get_metrics(projects: Sequence[Project]) -> Sequence[MetricMeta]:
             org_id=projects[0].organization_id,
         )
     )
+    metric_names_meta_gen = (
```

Comment:
I quite dislike how this function is written... just write a for-loop that appends to a list! You apparently need a sorted list later anyway.